sdks/python/setup.py

```diff
@@ -30,9 +28,15 @@
 from pkg_resources import DistributionNotFound
 from pkg_resources import get_distribution
 from pkg_resources import normalize_path
+from pkg_resources import parse_version
 from pkg_resources import to_filename
 from setuptools import Command
 
+# distutils from stdlib is deprecated from python 3.10.
+# setuptools has been maintaining a copy of distutils and importing
+# setuptools will replace stdlib/distutils with setuptools/distutils.
+from distutils.errors import DistutilsError # pylint: disable=wrong-import-order
```

Comment:
So basically, as long as setuptools is always imported before distutils, we can still use `DistutilsError` because it's maintained separately inside `setuptools`?

Reply:
And we would rather keep this because `DistutilsError` is more meaningful than your previously proposed usage of `setuptools.errors.BaseError`?

Reply:
So the `distutils` is present in two places. Python standard library and also in setuptools. `setuptools` adopted stdlib implementation of `distutils` and the `stdlib distutils` is marked for deprecation  in Python 3.10 and removal for Python 3.12.

Now, in order to use `distutils`, we can just import `setuptools`, which implicitly imports its version of `distutils`. As of now, there are two versions of `distutils`, it is recommended to import `distutils` after `setuptools`.

I read a bit and found the information above. Now we don't need the `BaseError` which requires `setuptools>=60.0.0`. We just need to reorder the imports wrt `setuptools` and `distutils`.

Reply:
Ah, that makes sense. Now we don't need to try to maintain `setuptools>=60.0.0` requirement. Thanks for the clarification.

Reply:
> Now we don't need to try to maintain setuptools>=60.0.0 requirement.
Sounds like we need v48 or higher though.